MetaMorpheus/EngineLayer/SpectralLibrarySearch/CrosslinkLibrarySpectrum.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Proteomics.Fragmentation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using EngineLayer.CrosslinkSearch;
+using System.Threading.Tasks;
+
+namespace EngineLayer
+{
+    public class CrosslinkLibrarySpectrum : LibrarySpectrum
+    {
+        public CrosslinkLibrarySpectrum BetaPeptideSpectrum { get; }
+        public string AlphaPeptideSequence { get; private set; }
+        public string BetaPeptideSequence { get; private set; }
+        public bool IsBetaPeptide { get; }
+        public static Regex CrosslinkRegex = new Regex(@"\(\d+\)");
+
+        public CrosslinkLibrarySpectrum(
+            string sequence,
+            double precursorMz,
+            int precursorCharge,
+            List<MatchedFragmentIon> peaks,
+            double rt,
+            CrosslinkSpectralMatch betaPeptide,
+            bool isDecoy = false) : base(sequence, precursorMz, precursorCharge, peaks, rt, isDecoy)
+        {
```

Comment:
is it possible for this constructor to call this(sequence, precursorMz, precursorCharge, betaPeptide.MatchedFragmentIons, rt, new CrosslinkLibrarySpectrum(....), isDecoy) instead of having multiple constructors with similar logic?

Reply:
Done